vault/ha.go

```diff
@@ -462,6 +462,17 @@ func (c *Core) waitForLeadership(newLeaderCh chan func(), manualStepDownCh, stop
 		c.activeContext = activeCtx
 		c.activeContextCancelFunc.Store(activeCtxCancel)
 
+		// Perform seal migration before wiping the barrier/seal state
+		if err := c.migrateSeal(c.activeContext); err != nil {
+			c.heldHALock = nil
+			lock.Unlock()
+			close(continueCh)
+			c.stateLock.Unlock()
+			c.logger.Error("seal migration error", "error", err)
+			metrics.MeasureSince([]string{"core", "leadership_setup_failed"}, activeTime)
+			continue
```

Comment:
The other invocation of `c.migrateSeal` seals itself on failure, I wonder if we want to do that here as well?

Reply:
The reason I left it this way is to be consistent with how we are behaving when `postUnseal` fails or if `setupCluster` fails. I think this is okay. Let me know.

Reply:
If we fail a migration the safest thing to do is probably seal, I'd vote we change it.

Reply:
Done. But please look at the change. It would be good to know if it looks okay or if there is anything else to be done.

Reply:
I suspect we will want to return from this function instead of continue?